project-koku/koku
# Sources remove promotion, and filter unknown edits

While testing Sources in CI a two situations came up that we should handle differently.

Issue 1
In a previous change I added logic to "promote" a koku provider to a source if the same billing_source and authentication were used to add the source.  This logic added unnecessary complexity for a situation that is very limited.  That situation is the ability for any existing Koku-providers to be added to platform sources before full UI integration is complete.

This operation can still be accomplished with a manual database table update if necessary.

Issue 2
There were several sources added to platform sources before cost management's integration.  Because of this we missed the initial `Application.create` event and the headers being used are not configured to be used with cost management.  A `Source.update` filter is added to only proceed on sources that we know about.  We can reach out to the source owners to either get them to remove and re-add cost management if they want cost management attached to their source.

**Testing**
1.  Add a koku provider and attempt to add same account to platform sources.  Verify that the source creation fails.
2.  Add a source via platform sources and verify it's successful.  Delete the entry from the sources db table and attemp to edit the source.  Verify Source.update message is ignored.

**Test Results**
[sources_ci_updates_ut.txt](https://github.com/project-koku/koku/files/3888302/sources_ci_updates_ut.txt)
